lib/barlix/gs1.ex

```diff
@@ -0,0 +1,194 @@
+defmodule Barlix.GS1 do
+  @moduledoc """
+  This module implements the [GS1-128](https://en.wikipedia.org/wiki/GS1-128) symbology with Code Set C only. Only FNC-1 and pair of digits are accepted
+  """
+
+  @doc """
+  Encodes the given value using GS1 128 symbology.
+
+  Must have pair of digits and optionally Function Code 1 (:fnc_1)
+
+  Examples
+
+      Barlix.GS1.encode("123456")
+
+      Barlix.GS1.encode([:fnc_1, ?1, ?2, ?3, ?4, :fnc_1, ?5, ?6])
+  """
+  @spec encode(String.t() | [member]) :: {:error, binary} | {:ok, Barlix.code()}
+        when member: char() | :fnc_1
+  def encode(value) do
+    value = Barlix.Utils.normalize_string(value)
+
+    with :ok <- validate(value),
+         do: loop(value)
+  end
+
+  @doc """
+  Accepts the same arguments as `encode/1`. Returns `t:Barlix.code/0` or
+  raises `Barlix.Error` in case of invalid value.
+  """
+  @spec encode!(String.t() | [char() | :fnc_1]) :: Barlix.code() | no_return
```

Comment:
Will likely need to fix this typespec too

Reply:
damn it! 🤦